### PR TITLE
Add some functions

### PR DESCRIPTION
**Pull request by hidehiro98**
@db0sch 🙏

### CODE REVIEW

File: app/controllers/requests/received_requests_controller.rb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 class Requests::ReceivedRequestsController < ApplicationController
   def index
-    @requests = current_user.received_requests
+    @requests = current_user.received_requests.sort_by { |request| request.duedate }
```
Comment (db0sch) [inline app/controllers/requests/received_requests_controller.rb]: you should use a scope. cf active record query documentation

File: app/controllers/requests/sent_requests_controller.rb [hunk 2]
```diff
@@ -1,5 +1,5 @@
 class Requests::SentRequestsController < ApplicationController
   def index
-    @requests = current_user.requests
+    @requests = current_user.requests.sort_by { |request| request.duedate }
```
Comment (db0sch) [inline app/controllers/requests/sent_requests_controller.rb]: same here.

File: app/models/request.rb [hunk 3]
```diff
@@ -7,6 +7,7 @@ class Request < ApplicationRecord
   validates :content, :user, :duedate, presence: true
 
   enum status: { pending: 0, accepted: 1, rejected: 2, commented: 3 }
+  enum open_closed: { open: 0, closed: 1 }
```
Comment (db0sch) [inline app/models/request.rb]: i think, you can add that to the status enum.
`{ pending: 0, accepted: 1, rejected: 2, commented: 3, closed: 4 }`

0, 1, 2, and 3 are open.
4 is closed.

File: db/migrate/20170619114606_add_open_closed_to_request.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+class AddOpenClosedToRequest < ActiveRecord::Migration[5.0]
+  def change
+    add_column :requests, :open_closed, :integer, default: 0
```
Comment (db0sch) [inline db/migrate/20170619114606_add_open_closed_to_request.rb]: no need. cf above.

**Review by db0sch — CHANGES_REQUESTED** [review]

File: app/controllers/requests/received_requests_controller.rb (hunk 1 above)
Comment (hidehiro98) [reply app/controllers/requests/received_requests_controller.rb]: I found it. thank you!
http://guides.rubyonrails.org/active_record_querying.html#ordering

File: app/models/request.rb (hunk 3 above)
Comment (hidehiro98) [reply app/models/request.rb]: I thought about that, but if I did so, user cannot use open/closed and pending/accepted... at the same time as a search query.
I think it is not convenient, how do you think?

Comment (db0sch) [reply app/models/request.rb]:
```
enum status: { pending: 0, accepted: 1, rejected: 2, commented: 3, canceled: 4 }
```
with

```
scope :open, -> { where(status: [:pending, :commented]) }
scope :closed, -> { where(status: [:accepted, :rejected, :canceled]) }
```

Comment (hidehiro98) [reply app/models/request.rb]: Thank you very much!!

It means when request accepted, it's automatically closed, right?
Before, I thought user close requests by click 'close' button, but I think automatic is better.
So I also changed the tabs 'Open / To close / Closed' to 'Open / Closed', 'Close' button in cards to 'Cancel' button.

**Review by db0sch — APPROVED** [review]
LGTM

**Comment by hidehiro98** [comment]
ありがとう！

**Comment by db0sch** [comment]
:fist_oncoming:

**Pull request merged**